File: examples/awfy/node/harness.mjs
export function runBench(b) {
  let warmSink = 0;
  for (let w = 0; w < b.warmup; w++) warmSink ^= b.run(b.size);
  if (warmSink === 0x7fffffff) console.log("unreachable", warmSink);

  const start = performance.now();
  let checksum = 0;
  for (let i = 0; i < b.iters; i++) checksum = b.run(b.size);
  const elapsed = performance.now() - start;

  if (checksum !== b.expected) {
    throw new Error(`${b.name}: checksum ${checksum} != expected ${b.expected}`);
  }
  console.log(`node\t${b.name}\t${b.iters}\t${elapsed}\t${checksum}`);
}

File: examples/awfy/go/harness.go
package main

import (
	"fmt"
	"time"
)

type Bench struct {
	Name     string
	Warmup   int
	Iters    int
	Size     int
	Expected int
	Run      func(int) int
}

func RunBench(b Bench) {
	warmSink := 0
	for w := 0; w < b.Warmup; w++ {
		warmSink ^= b.Run(b.Size)
	}
	if warmSink == 0x7fffffff {
		fmt.Println("unreachable", warmSink)
	}

	start := time.Now()
	checksum := 0
	for i := 0; i < b.Iters; i++ {
		checksum = b.Run(b.Size)
	}
	elapsedMs := float64(time.Since(start).Nanoseconds()) / 1e6

	if checksum != b.Expected {
		panic(fmt.Sprintf("%s: checksum %d != expected %d", b.Name, checksum, b.Expected))
	}
	fmt.Printf("go\t%s\t%d\t%g\t%d\n", b.Name, b.Iters, elapsedMs, checksum)
}
